core/src/main/java/feign/DefaultMethodHandler.java

```
@@ -1,114 +1,164 @@
-    Class<?> declaringClass = defaultMethod.getDeclaringClass();
-
-    try {
-      Lookup lookup = readLookup(declaringClass);
-      this.unboundHandle = lookup.unreflectSpecial(defaultMethod, declaringClass);
-    } catch (NoSuchFieldException | IllegalAccessException | IllegalArgumentException
-        | InvocationTargetException ex) {
-      throw new IllegalStateException(ex);
+    // Uses Java 7 MethodHandle based reflection. As default methods will only exist when
+    // run on a Java 8 JVM this will not affect use on legacy JVMs.
+    // When Feign upgrades to Java 7, remove the @IgnoreJRERequirement annotation.
+    public MethodHandle unboundHandle;
+
+    // handle is effectively final after bindTo has been called.
+    private MethodHandle handle;
+
+
+    public DefaultMethodHandler(Method defaultMethod) {
+        /**
+         * 13/06/2021
+         *
+         * Modified to work with Android. Previous commit worked up to API level 28 (Android 8)
+         * however, Android 9+ enforces more strict checking with reflection.
+         *
+         * Solution is to use double reflection, as below.
+         *
+         * Jamie (xrpdevs-at-xrpdevs-co-uk)
+         *
+         **/
+
+        Class<?> declaringClass = defaultMethod.getDeclaringClass();
+
+        try {
+            if (System.getProperty("java.vm.name").equalsIgnoreCase("Dalvik")) {
+                androidObtainHandle(defaultMethod, declaringClass);
+            } else {
+                Lookup lookup = readLookup(declaringClass);
+                this.unboundHandle = lookup.unreflectSpecial(defaultMethod, declaringClass);
+            }
+
+        } catch (IllegalAccessException ex) {
+            throw new IllegalStateException(ex);
+        } catch (InstantiationException | InvocationTargetException | NoSuchMethodException e) {
+            e.printStackTrace();
+        } catch (NoSuchFieldException e) {
+            e.printStackTrace();
```

this is not merge-able, as this is eating the exceptions.  If we are unable to get the `unboundHandle` instance, throw the exception and let it blow.

Since you are going to touch this, please also include an message to this exception

---

Currently I'm building this with Gradle in Android Studio.

Will refactor tomorrow and try to build in Maven against different JDKs.

---

See latest commit. Hopefully this fixes problems you mentioned

---

I will wait until build passes to give this a second look